File: project_euler/problem4.rb
def is_palindrome?(num)
  i = 0
  number_string = num.to_s
  while i < number_string.length
    if num[i] == num[num.length-1-i]
    	return false
    end
  end
  return true
end

def products
 palindromes = []
 
 i = 100
 while i < 1000
  j = 100
  while j < 1000
    if is_palindrome?(i*j)
    	palindrome << i*j
    end
  	j += 1
  end
 	i += 1
 end

 palindrome.max
end

File: problem4.py
def is_palindrome(num):
  string = str(num)
  i = 0
  while (i < len(string)):
    if (string[i] != string[len(string) - i - 1]):
      return False
    i += 1
  return True

def largest_product():
  i = 999
  result = 0
  while (i > 0):
    j = i
    while (j > 0):
      if (is_palindrome(i * j) and i *j > result):
        result = i * j
      j -= 1
    i -= 1
  return result

print largest_product()
